src/main/java/sirius/db/jdbc/SmartQuery.java

```diff
@@ -320,8 +318,44 @@ protected Iterator<E> pullNextBlock() {
             return block.iterator();
         }
 
+        private SmartQuery<E> adjustQuery(SmartQuery<E> query) {
+            SmartQuery<E> adjusted = query.copy();
+
+            if (adjusted.limit > 0) {
+                throw new UnsupportedOperationException("SmartQuery doesn't allow 'limit' in streamBlockwise");
+            }
+            if (adjusted.skip > 0) {
+                throw new UnsupportedOperationException("SmartQuery doesn't allow 'skip' in streamBlockwise");
+            }
+
+            // we need to guarantee an absolute ordering
+            if (adjusted.distinct) {
+                // we have distinct fields, so we can easily create an absolute ordering, if it's not already there
+                if (!new HashSet<>(Tuple.firsts(adjusted.orderBys)).containsAll(adjusted.fields)) {
+                    Set<Mapping> missingOrderBys = new HashSet<>(adjusted.fields);
+                    Tuple.firsts(orderBys).forEach(missingOrderBys::remove);
+                    missingOrderBys.forEach(adjusted::orderAsc);
+                }
```

Comment:
couldn't this be fixed in a single stream pipeline - should kinda be possible (?)

Reply:
is this ordering guaranteed to be absolute? shouldn't the id still be there just in case?

Reply:
The query is `distinct`, so it should suffice to ORDER BY all the selected fields